README.pod
=pod

=encoding utf8

=head1 NAME

Alien::HIDAPI - Perl distribution for HIDAPI

=head1 INSTALL

    cpan Alien::HIDAPI

=head1 PLATFORMS
    
Same as HIDAPI proper: Linux, FreeBSD, macOS and Windows.

=head1 GIT REPOSITORY

L<http://github.com/athreef/Alien-HIDAPI>

=head1 SEE ALSO

L<HIDAPI|http://github.com/signal11/hidapi>

L<Alien>


=head1 AUTHOR

Ahmad Fatoum C<< <athreef@cpan.org> >>, L<http://a3f.at>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2017 Ahmad Fatoum

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut


CHANGES
==================================================
Changes from 2017-06-25 13:30:41 +0000 to present.
==================================================

------------------------------------------
version 0.004 at 2017-07-05 17:48:32 +0000
------------------------------------------

  Change: bf036e76289aea0015ddf0e9a8c6990da6fd2cd8
  Author: Ahmad Fatoum <ahmad@a3f.at>
  Date:   2017-07-05 19:48:32 2017 +0200

    Linux: Fetch Alien:: packages for libusb1/libudev

------------------------------------------
version 0.003 at 2017-07-04 03:34:42 +0000
------------------------------------------

  Change: cf3e225e620acbd6b9e792a92dd396ed614edbd4
  Author: Ahmad Fatoum <ahmad@a3f.at>
  Date : 2017-07-04 05:34:42 +0200
 
    Install Alien:Autotools if not available
 
------------------------------------------
version 0.002 at 2017-07-01 22:58:06 +0000
------------------------------------------
 
  Change: e33b61800aefe03e5d5715401bc3e6ca7558b858
  Author: Ahmad Fatoum <ahmad@a3f.at>
  Date : 2017-07-02 00:58:06 +0200
 
    Require IO::Socket::SSL, so Github download suceeds
    
    There should be a HTTP fallback instead, but this is easier
 
------------------------------------------
version 0.001 at 2017-06-25 13:30:41 +0000
------------------------------------------
 
  Change: 31ae276e629dbcbbae62fd227f4e413c96edbb5d
  Author: Ahmad Fatoum <ahmad@a3f.at>
  Date : 2017-06-25 15:30:41 +0200
 
    Initial commit 
 
================
End of releases.
================


Build.PL
use Alien::Base::ModuleBuild;
use Config;
use Alien::autoconf;
use Alien::automake;
use Alien::libtool;
use Alien::libudev;
use Alien::LibUSBx;

use Env qw( @PATH );

unshift @ENV, Alien::automake->bin_dir, Alien::autoconf->bin_dir, Alien::libtool->bin_dir;

my $build = Alien::Base::ModuleBuild->new(
    module_name => 'Alien::HIDAPI',
    license => 'perl',
    alien_bin_requires => {
    },
    configure_requires => {
        'Alien::Base::ModuleBuild' => 0,
        'Module::Build' => 0,
    },
    build_requires => {
        # TODO: fallback to HTTP download when IO::Socket::SSL doesn't exist
        'IO::Socket::SSL' => 0,

        "Alien::autoconf" => "0.02",
        "Alien::automake" => "0.02",
        "Alien::libtool" => "0.02",

        $^O eq 'linux' ? ('Alien::LibUSBx' => 0,
                          'Alien::libudev' => 0)
                       : (),
    },
    recommends => {
        'IO::Socket::SSL' => 0
    },
    test_requires => {
        'Test::More'         => 0,
        'ExtUtils::CBuilder' => 0,
        'Test::Alien'        => '0.05',
        'Test2'              => '1.302015',
        'Test2::Suite'       => '0.000030',
    },
    requires => {
        'perl' => '5.008001',
        'Alien::Base' => '0.030',
        $^O eq 'linux' ? ('Alien::LibUSBx' => 0,
                          'Alien::libudev' => 0)
                       : (),
    },
    dist_author => 'Ahmad Fatoum <athreef@cpan.org>',
    alien_name => 'HIDAPI',
    alien_repository => {
        protocol => 'http',
        host     => 'github.com',
        location => '/signal11/hidapi/archive',
        exact_filename => 'master.tar.gz',
    },
    alien_inline_auto_include => [ 'hidapi.h' ],
    alien_stage_install => 1,

    alien_build_commands => [
        './bootstrap',
        '%c --prefix=%s',
        'make',
    ],
    alien_install_commands => ['make install'],
    extra_compiler_flags => [Alien::libudev->config('cflags'), Alien::LibUSBx->config('cflags')],
    extra_linker_flags => [Alien::libudev->config('libs'), Alien::LibUSBx->config('libs')],
);

$build->create_build_script;
#$build->dispatch('distmeta');
#$build->dispatch('manifest');
#$build->dispatch('build');


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Perl distribution for HIDAPI",
  "AUTHOR" => "Ahmad Fatoum <athreef\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Base" => "0.038",
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "Alien::Build::Plugin::Build::Autoconf" => 0,
    "Alien::Build::Plugin::Gather::IsolateDynamic" => 0,
    "Alien::MSYS" => 0,
    "Alien::autoconf" => "0.02",
    "Alien::automake" => "0.02",
    "Alien::libtool" => "0.02",
    "Config" => 0,
    "ExtUtils::MakeMaker" => "6.52"
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "DISTNAME" => "Alien-HIDAPI",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008001",
  "NAME" => "Alien::HIDAPI",
  "PREREQ_PM" => {
    "Alien::Base" => "0.038",
    "base" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.52",
    "File::Spec" => 0,
    "Test2::Suite" => 0,
    "Test2::V0" => 0,
    "Test::Alien" => "0.05",
    "Test::More" => "0.88"
  },
  "VERSION" => "0.001",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Alien::Base" => "0.038",
  "Alien::Build" => "0.32",
  "Alien::Build::MM" => "0.32",
  "Alien::Build::Plugin::Build::Autoconf" => 0,
  "Alien::Build::Plugin::Gather::IsolateDynamic" => 0,
  "Alien::MSYS" => 0,
  "Alien::autoconf" => "0.02",
  "Alien::automake" => "0.02",
  "Alien::libtool" => "0.02",
  "Config" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "File::Spec" => 0,
  "Test2::Suite" => 0,
  "Test2::V0" => 0,
  "Test::Alien" => "0.05",
  "Test::More" => "0.88",
  "base" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
sub MY::postamble {
  $abmm->mm_postamble;
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild

